Tests/Realm.Tests/Sync/SyncTestHelpers.cs

```diff
@@ -118,7 +126,14 @@ public static async Task TriggerClientResetOnServer(SyncConfigurationBase config
             }
 
             var result = await config.User.Functions.CallAsync<BaasClient.FunctionReturn>("triggerClientResetOnSyncServer", userId, appId);
-            Assert.That(result.status, Is.EqualTo(BaasClient.FunctionReturn.Result.success));
+            if (result.Deleted > 0)
+            {
+                // This is kind of a hack, but it appears like there's a race condition on the server, where the deletion might not be
+                // registered and the server will not respond with a client reset. Doing the request again gives the server some extra time
+                // to process the deletion.
```

Comment:
I think it's just an inherent race condition between deleting the file and establishing the session. The extra http call is a silly way to do it - I could also just do a `Task.Delay` if you think it'll be more sensible.

Reply:
The result it's the same. I was mostly interested in knowing if this was something somebody on their team was gonna look into. Since you believe it's an **inherent** issue, then there's no bug to fix.